nat-lab/tests/uniffi/libtelio_remote.py

```diff
@@ -51,38 +52,40 @@ def next_event(self) -> libtelio.Event:
 
 
 class TelioLoggerCbImpl(libtelio.TelioLoggerCb):
-    def __init__(self):
+    def __init__(self, logfile):
         self.lock = Lock()
-        try:
-            os.remove(TCLI_LOG)
-        except FileNotFoundError:
-            pass
+        self.logfile = logfile
 
-    def log(self, log_level, payload):
+    def _do_log(self, log_level, payload):
+        write_ts = None
+        start_ts = time.time()
         with self.lock:
-            with open(TCLI_LOG, "a", encoding="utf-8") as logfile:
-                try:
-                    logfile.write(f"{datetime.datetime.now()} {log_level} {payload}\n")
-                except IOError as e:
-                    logfile.write(
-                        f"{datetime.datetime.now()} Failed to write logline due to error {str(e)}\n"
-                    )
+            lock_acquired_ts = time.time()
+            self.logfile.write(f"{datetime.datetime.now()} {log_level} {payload}\n")
+            write_ts = time.time()
+        end_ts = time.time()
```

Comment:
nit: I'm not sure we need `write_ts` if we already have `end_ts`

Reply:
Initially I added it just to check whether unlocking is not causing delays. But it does not, and I guess it is reasonably safe to assume, that it will not.

Hence: Fixed